java/arcs/android/storage/service/IDevToolsProxy.aidl

```diff
@@ -0,0 +1,23 @@
+package arcs.android.storage.service;
+
+import arcs.android.storage.service.IStorageServiceCallback;
+
+/**
+ * Exposed API to communicate between [StorageService] and [DevToolsService].
+ */
+interface IDevToolsProxy {
+
+    /**
+     * TODO: (sarahheimlich) remove once we dive into stores (b/162955831)
+     *
+     * Register a callback to be called with the [BindingContext] receives a [ProxyMessage]
+     */
+    oneway void registerBindingContextProxyMessageCallback(in IStorageServiceCallback callback);
+
+    /**
+     * TODO: (sarahheimlich) remove once we dive into stores (b/162955831)
+     *
+     * Remove a callback that is called with the [BindingContext] receives a [ProxyMessage]
+     */
+    oneway void deRegisterBindingContextProxyMessageCallback(in IStorageServiceCallback callback);
```

Comment:
Have you verified that register/deRegister work? I'm suspicious that the binder versions of the callbacks will be able to be identified with eachother after going over a process boundary.  ie. callbackA may be the same exact object (or at least resolve to match itself with `.equals(..)`) on the client-side, but after being transformed into an IPC binder object once during `registerBindingContextProxyMessageCallback`, I'm not sure that the result of transforming it again in `deRegisterBindingContextProxyMessageCallback` would necessarily maintain that `callbackA` from the first is equal to `callbackB` from the second on the service-side.

This is one of the reasons we return identifiers in the IStorageService interface for adding callbacks, so we can pass that identifier on the unregister-call instead.


It's fine to keep the way you have if you know that the way the service will add and remove the callback binders will be able to work correctly. I'm not sure an automated test would work here, unless it were an instrumentation test - because it's not possible to have two processes going during a simple unit test.

Reply:
I tried it and it failed, added callback tokens and manually tested and it worked.